src/core/Tokenizer.js

```diff
@@ -165,23 +173,33 @@ export default class Tokenizer {
         });
     }
 
-    getVariableToken(input) {
-        return this.getPlainVariableToken(input) || this.getQuotedVariableToken(input);
+    getPlaceholderToken(input) {
+        return this.getPlainNamedPlaceholderToken(input) ||
+            this.getStringNamedPlaceholderToken(input) ||
+            this.getIndexedPlaceholderToken(input);
+    }
+
+    getPlainNamedPlaceholderToken(input) {
+        return this.getTokenOnFirstMatch({
+            input,
+            type: sqlTokenTypes.PLACEHOLDER,
+            regex: this.PLAIN_NAMED_PLACEHOLDER_REGEX
+        });
     }
 
-    getPlainVariableToken(input) {
+    getStringNamedPlaceholderToken(input) {
         return this.getTokenOnFirstMatch({
             input,
-            type: sqlTokenTypes.VARIABLE,
-            regex: this.PLAIN_VARIABLE_REGEX
+            type: sqlTokenTypes.PLACEHOLDER,
+            regex: this.STRING_NAMED_PLACEHOLDER_REGEX
         });
     }
 
-    getQuotedVariableToken(input) {
+    getIndexedPlaceholderToken(input) {
         return this.getTokenOnFirstMatch({
             input,
-            type: sqlTokenTypes.VARIABLE,
-            regex: this.QUOTED_VARIABLE_REGEX
+            type: sqlTokenTypes.PLACEHOLDER,
+            regex: this.INDEXED_PLACEHOLDER_REGEX
         });
     }
```

Comment:
We could generalize these three methods - the only difference between them is the regex.